src/renderer/features/proxies/model/proxies-model.ts

```diff
@@ -157,10 +189,80 @@ sample({
 
 sample({
   clock: getProxiesFx.done,
-  fn: ({ params: { chainId } }) => chainId,
+  source: $endpoint,
+  filter: (endpoint) => Boolean(endpoint),
+  fn: (endpoint, { params: { chainId } }) => ({ chainId, endpoint: endpoint! }),
   target: disconnectFx,
 });
 
+sample({
+  clock: depositsReceived,
+  source: {
+    wallets: walletModel.$wallets,
+    accounts: walletModel.$accounts,
+    proxyChainGroups: proxyModel.$proxyChainGroups,
+  },
+  fn: ({ wallets, accounts, proxyChainGroups }, deposits) => {
```

Comment:
I guess something like this would be much more readable
Probably moving `const { toAdd, toUpdate }` & `const toRemove` to utils would be great as well 🤔
```javascript
  fn: ({ wallets, accounts, proxyChainGroups }, deposits) => {
    const proxyGroups = proxiesUtils.getprogyGroups(wallets, accounts);

    const { toAdd, toUpdate } = proxyGroups.reduce<Record<'toAdd' | 'toUpdate', NoID<ProxyChainGroup>[]>>((acc, g) => {
      const shouldAddAll = proxyChainGroups.every((p) => proxyUtils.isSameProxyChainGroup(p, g));

      if (shouldAddAll) {
        acc.toAdd.push(g);
      } else {
        acc.toUpdate.push(g);
      }

      return acc;
    }, { toAdd: [], toUpdate: [] });

    const toRemove = proxyChainGroups.filter((p) => {
      if (p.chainId !== deposits.chainId) return false;

      return proxyGroups.every((g) => proxyUtils.isSameProxyChainGroup(g, p));
    });

    return { toAdd, toUpdate, toRemove };
  }
```